docs/highlighters/regexp.md

```diff
@@ -0,0 +1,24 @@
+zsh-syntax-highlighting / highlighters / regexp
+------------------------------------------------
+
+This is the `regexp` highlighter, that highlights user-defined regular expressions. It's similar to pattern highlighter, but allows more complex patterns.
```

Comment:
Please wrap prose paragraphs to 80 chars, since the markdown source might be read by `$PAGER` in a terminal.